edb/pgsql/delta.py

```diff
@@ -4850,59 +4865,60 @@ def _compile_conversion_expr(
                 orig_schema, pointer, env=None
             )
 
-        refs = irutils.get_longest_paths(ir.expr)
-        ref_tables = schemamech.get_ref_storage_info(ir.schema, refs)
-        local_table_only = all(
-            t == old_ptr_stor_info.table_name
-            for t in ref_tables
-        )
+        # refs = irutils.get_longest_paths(ir.expr)
+        # ref_tables = schemamech.get_ref_storage_info(ir.schema, refs)
+        # local_table_only = all(
+        #     t == old_ptr_stor_info.table_name
+        #     for t in ref_tables
+        # )
 
         ptr_path_id = tgt_path_id.ptr_path()
         src_path_id = ptr_path_id.src_path()
         assert src_path_id
 
-        external_rels = {}
-        external_rvars = {}
+        src_rel = compiler.new_external_rel(
+            rel_name=(source_alias,),
+            path_id=src_path_id,
+        )
+        src_rel.path_outputs[(src_path_id, 'iterator')] = \
+            pgast.ColumnRef(name=('id',))
+        external_rels = {
+            src_path_id: (src_rel, ('source', 'identity', 'iterator'))
+        }
 
-        if produce_ctes:
-            external_rels[src_path_id] = compiler.new_external_rel(
-                rel_name=(source_alias,),
-                path_id=src_path_id,
-            ), ('value', 'source')
-        else:
+        if for_each_pointer:
             if ptr_table:
-                rvar = compiler.new_external_rvar(
-                    rel_name=(source_alias,),
-                    path_id=ptr_path_id,
-                    outputs={
-                        (src_path_id, ('identity',)): 'source',
-                    },
-                )
-                external_rvars[ptr_path_id, 'source'] = rvar
-                external_rvars[ptr_path_id, 'value'] = rvar
-                external_rvars[src_path_id, 'identity'] = rvar
-                external_rvars[tgt_path_id, 'identity'] = rvar
-                if local_table_only and not is_lprop:
-                    external_rvars[src_path_id, 'source'] = rvar
-                    external_rvars[src_path_id, 'value'] = rvar
-                elif is_lprop:
-                    external_rvars[tgt_path_id, 'value'] = rvar
+                src_rel.path_outputs[(ptr_path_id, 'identity')] = \
```

Comment:
Hm, I'm a little nervous about making ctid an `identity` aspect; what needs that?